### PR TITLE
use validator for nock payload validation

### PR DESCRIPTION
**Pull request by mikicho**
Is it better?

### CODE REVIEW

**Review by goldbergyoni — COMMENTED** [review]
@mikicho I'm split, I would ask people in a group, maybe? Let's use a less scary regex in the comparison?

File: various-receipes/3rd-party-services-isolation/test/isolation.test.js [hunk 1]
```diff
@@ -95,11 +96,10 @@ describe("/api", () => {
       // ️️️✅ Best Practice: Intercept requests for 3rd party services to eliminate undesired side effects like emails or SMS
       // ️️️✅ Best Practice: Specify the body when you need to make sure you call the 3rd party service as expected 
       const scope = nock("https://mailer.com")
-        .post('/send', {
-          subject: /^(?!\s*$).+/, 
-          body: /^(?!\s*$).+/, 
-          recipientAddress: /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/,
-        })
+        .post('/send', payload => validator.isEmail(payload.recipientAddress)
```
[the remark this answers is not in the document]
Comment (mikicho) [reply various-receipes/3rd-party-services-isolation/test/isolation.test.js:99]: I'll check the "catch-all -> expect" approach.

[the remark this answers is not in the document]
Comment (DanielGluskin) [reply various-receipes/3rd-party-services-isolation/test/isolation.test.js:99]: any posibility to write something like:
```
body: validator.isString,
email: validator.isEmail
```

if not, I actually prefer the previous option.

**Comment by goldbergyoni** [comment]
@mikicho @DanielGluskin 

How about:

```
test('When adding a new user, Then an email should be sent', () => {
  // Arrange
  const user = {name: 'Michael'}
  let emailRequest;
  nock('http://email.com').post('/new').reply(200, (url, request) => {
    emailRequest = request;
  })
  
  // Act
  axios.post('/user', user);
  
  // Assert
  expect(emailRequest).toMatchObject({
    subject: expect.any(String),
    body: expect.stringMatching(/welcome|user/),
  });
});
```

(Never executed it, from the top of my head)

**Comment by mikicho** [comment]
@goldbergyoni 
wow! this is both elegant and ugly at the same time! :D 
But... I really think this is magicless and gets the job done. so I totally agree with this style! 
Great idea!!

p.s: it will work if the function will return true.

**Comment by goldbergyoni** [comment]
Copyright: Friend Commissar @Rashe

**Comment by mikicho** [comment]
@goldbergyoni @DanielGluskin 
We can write:
```
nock("https://mailer.com")
        .post('/send', payload => (emailPayload = payload, true)
        .reply(202);
```

but didn't want to confuse.

**Comment by goldbergyoni** [comment]
> but didn't want to confuse.

It's actually interesting... and shorter

**Review by DanielGluskin — APPROVED** [review]
very nice. maybe can be switched to a one-liner:
`.post('/send', payload => emailPayload = payload)`

**Comment by goldbergyoni** [comment]
@mikicho I love your shortened version

**Comment by mikicho** [comment]
@goldbergyoni 
@DanielGluskin version works.
I didn't suggest that because I'm afraid we rely on nock's internal implementation. 
WDYT?

**Comment by goldbergyoni** [comment]
@mikicho What is nock implementation? 

What @DanielGluskin wrote seems like public API to me:

```
.post('/send', payload => emailPayload = payload)
```

**Comment by mikicho** [comment]
@goldbergyoni the function needs to return true/false. but they probably doing something like:
```
const isIntercenptorSatisfied = ourFunction(body);
if (isIntercenptorSatisfied ) {
 ...
 }
```

but I'm afraid that maybe one day it will be:
```
const isIntercenptorSatisfied = ourFunction(body);
if (isIntercenptorSatisfied === true ) {
 ...
 }
```

**Review by goldbergyoni — APPROVED** [review]
LGTM 🌈💚

**Pull request merged**